### ISSUE TITLE
reportUnknownMemberType not logged when inside dict.update

### ISSUE THREAD

**Issue by mehdigmira** [microsoft/pyright#6634]
**Describe the bug**
A recent change in pyright seems to make reportUnknownMemberType errors disappear when an untyped method is used inside a dict.update call. I have tried to read recent release notes, but can't find smth related to this, so I think it's a bug.

**Code or Screenshots**

```python
# pyright: strict

from __future__ import annotations

from typing import Any


class A:
    def f(self, *args)->Any:  
        ...


A().f(1, 2)  # reportUnknownMemberType logged

d: dict[str, Any]={}
d.update({"foo": A().f(1, 2)})  # no error
```

**VS Code extension or command-line**
pyright 1.1.338

**Comment by erictraut** [comment]
Thanks for the bug report. This was a regression related to [this issue](https://github.com/microsoft/pyright/issues/6527).

This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.339, which I just published. It will also be included in a future release of pylance.
